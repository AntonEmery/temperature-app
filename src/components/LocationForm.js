import React, { Component } from 'react';


class LocationForm extends Component {
  constructor(props) {
    super(props)

    this.state = {
      location: 'San Francisco'
    }
  }

  handleChange = (e) => {
    this.setState({ location: e.target.value })
  }

  handleSubmit = (e) => {
    this.props.submitLocation(this.state.location);
    e.preventDefault();
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="location">Location
        <select
          name="location"
          value={this.state.location}
          onChange={this.handleChange}
        >
          <option value="San Francisco">San Francisco, CA</option>
          <option value="Amsterdam">Amsterdam, Netherlands</option>
          <option value="Oakland">Oakland, CA</option>
          <option value="Rome">Rome, Italy</option>
          <option value="Cleveland">Cleveland, OH</option>
          <option value="Tel Aviv">Tel Aviv, Israel</option>
          <option value="New York">New York City, NY</option>
          <option value="Murmansk">Murmansk, Russia</option>
          <option value="Instanbul">Istanbul, Turkey</option>
        </select>

        </label>
        <input
          type="submit"
          value="Submit"
        />
      </form>
    )
  }
}

export default LocationForm