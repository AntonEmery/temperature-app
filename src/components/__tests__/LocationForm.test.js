import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import LocationForm from '../LocationForm';

configure({ adapter: new Adapter() });

describe('Location form component', () => {
  it('Component renders correctly', () => {
    const tree = renderer.create(<LocationForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Select input should change the state of form', () => {
    let form = shallow(<LocationForm />);
      form.find('select').simulate('change', { target: { value: 'Oakland' }})
      expect(form.state('location')).toEqual('Oakland')
  })
})




