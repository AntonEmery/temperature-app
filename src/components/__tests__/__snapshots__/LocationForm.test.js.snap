// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location form component Component renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="location"
  >
    Location
  </label>
  <select
    name="location"
    onChange={[Function]}
    value="San Francisco"
  >
    <option
      value="San Francisco"
    >
      San Francisco, CA
    </option>
    <option
      value="Amsterdam"
    >
      Amsterdam, Netherlands
    </option>
    <option
      value="Oakland"
    >
      Oakland, CA
    </option>
    <option
      value="Rome"
    >
      Rome, Italy
    </option>
    <option
      value="Cleveland"
    >
      Cleveland, OH
    </option>
    <option
      value="Tel Aviv"
    >
      Tel Aviv, Israel
    </option>
    <option
      value="New York"
    >
      New York City, NY
    </option>
    <option
      value="Murmansk"
    >
      Murmansk, Russia
    </option>
    <option
      value="Istanbul"
    >
      Istanbul, Turkey
    </option>
  </select>
  <p>
    <input
      className="btn"
      type="submit"
      value="Submit"
    />
  </p>
</form>
`;
